lib/sanity/serializable.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Sanity
+  # Serializable is responsible for configuring auto serialization or a default
+  # serializer. It also defines the default class serializer used when auto
+  # serialization is enabled.
+  #
+  # The auto_serialize macro is used to enable auto serialization.
+  #
+  # @example enables auto serialization
+  #   auto_serialize
+  #
+  # The serializer marco is used to define the default serializer.
+  #
+  # @example default to using a custom defined UserSerializer
+  #   serializer UserSerializer
+  #
+  module Serializable
+    class << self
+      def included(base)
+        base.extend(ClassMethods)
+      end
+    end
+
+    module ClassMethods
+      def default_serializer
+        @default_serializer ||= nil
+      end
+
+      private
+
+      def auto_serialize
+        @default_serializer = class_serializer
+      end
+
+      def serializer(serializer)
+        @default_serializer = serializer
+      end
+
+      def class_serializer
+        @class_serializer ||= Proc.new do |results|
+          results['result'].map do |result|
```

Comment:
I'm torn on this - it's tightly coupled and implicitly dependent upon 

1) Sanity returning "result" key/value pair 
2) Sanity::Http::Query returning the parsed JSON & result body

There's probably a better way to handle this in the long term. Fine for now.. but something we could improve upon.

Reply:
I'm not sure how you get around that. If they change the response that obviously breaks this, but wouldn't that apply to other parts like querying too? If they change how querying works things break as well.  At the end of the day, it's relatively easy to override the serializer which would allow anyone to fix it without much code until an update was made to the project. Unless you're thinking about something totally different?